server/src/main/java/org/elasticsearch/index/mapper/PassThroughObjectMapper.java

```diff
@@ -14,26 +14,37 @@
 import org.elasticsearch.xcontent.XContentBuilder;
 
 import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
 import java.util.Locale;
 import java.util.Map;
 
 import static org.elasticsearch.common.xcontent.support.XContentMapValues.nodeBooleanValue;
+import static org.elasticsearch.common.xcontent.support.XContentMapValues.nodeIntegerValue;
 
 /**
  * Mapper for pass-through objects.
  *
  * Pass-through objects allow creating fields inside them that can also be referenced directly in search queries.
  * They also include parameters that affect how nested fields get configured. For instance, if parameter "time_series_dimension"
  * is set, eligible subfields are marked as dimensions and keyword fields are additionally included in routing and tsid calculations.
+ *
+ * In case different pass-through objects contain subfields with the same name (excluding the pass-through prefix), their aliases conflict.
+ * To resolve this, the pass-through spec specifies which object takes precedence through required parameter "priority"; non-negative
+ * integer values are accepted, with the highest priority value winning in case of conflicting aliases.
  */
 public class PassThroughObjectMapper extends ObjectMapper {
     public static final String CONTENT_TYPE = "passthrough";
+    public static final String PRIORITY_PARAM_NAME = "priority";
 
     public static class Builder extends ObjectMapper.Builder {
 
         // Controls whether subfields are configured as time-series dimensions.
         protected Explicit<Boolean> timeSeriesDimensionSubFields = Explicit.IMPLICIT_FALSE;
 
+        // Controls which pass-through fields take precedence in case of conflicting aliases.
+        protected int priority = 0;
```

Comment:
Let's make this attribute required.

Reply:
I think it's cleaner to just make it required.